src/onegov/pay/collections/payment.py

```diff
@@ -86,31 +101,88 @@ def __eq__(self, other: object) -> bool:
         )
 
     def subset(self) -> Query[Payment]:
-        q = self.query().order_by(desc(Payment.created))
+        query = self.query()
+
+        # Filter by payment type
+        if self.payment_type == 'manual':
+            query = query.filter(Payment.provider_id.is_(None))
+        elif self.payment_type == 'provider':
+            query = query.filter(Payment.provider_id.isnot(None))
+
+        # Filter by payment status
+        if self.status:
+            query = query.filter(Payment.state == self.status)
 
-        if self.start:
-            q = q.filter(self.start <= Payment.created)
+        # Filter by payments by each associated ticket creation date
+        if self.ticket_start or self.ticket_end:
+            query = query.join(Ticket, Payment.id == Ticket.payment_id)
 
-        if self.end:
-            q = q.filter(Payment.created <= self.end)
+            if self.ticket_start:
+                query = query.filter(Ticket.created >= self.ticket_start)
+            if self.ticket_end:
+                query = query.filter(Ticket.created <= self.ticket_end)
 
-        q = q.options(joinedload(Payment.provider))  # type:ignore[misc]
-        q = q.options(undefer(Payment.created))
-        return q
+        return query.order_by(Payment.created.desc())
 
     @property
     def page_index(self) -> int:
         return self.page
 
-    def page_by_index(self, index: int) -> Self:
-        return self.__class__(self.session, self.source, index)
+    def page_by_index(self, index):
+        return self.__class__(
+            self.session,
+            page=index,
+            start=self.start,
+            end=self.end,
+            ticket_start=self.ticket_start,
+            ticket_end=self.ticket_end,
+            reservation_start=self.reservation_start,
+            reservation_end=self.reservation_end,
+            payment_type=self.payment_type,
+            source=self.source,
+            status=self.status
+        )
+
+    def tickets_by_batch(self) -> dict[UUID, Ticket]:
+        session = self.session
+        return {ticket.payment_id: ticket  # type:ignore[union-attr]
+            for ticket in session.query(Ticket).filter(
+            Ticket.payment_id.in_([el.id for el in self.batch]))
+        }
+
+    def reservation_dates_by_batch(self) -> dict[UUID, tuple[date, date]]:
+        from onegov.reservation import Reservation
+        session = self.session
+        return {
+            ticket_id: (
+                to_timezone(start_date, 'Europe/Zurich').date(),
+                to_timezone(end_date, 'Europe/Zurich').date()
+            )
+            for ticket_id, start_date, end_date in session.query(Reservation)
+            # .join(Ticket) didn't work
+            .join(Reservation.payment)
+            .join(Payment.tickets)
+            .filter(Ticket.payment_id.in_([el.id for el in self.batch]))
+            .group_by(Ticket.id)
+            .with_entities(
+                Ticket.id,
```

Comment:
```suggestion
            payment_id: (
                to_timezone(start_date, 'Europe/Zurich').date(),
                to_timezone(end_date, 'Europe/Zurich').date()
            )
            for payment_id, start_date, end_date in session.query(Reservation)
            .join(Reservation.payment)
            .filter(Payment.id.in_([el.id for el in self.batch]))
            .group_by(Payment.id)
            .with_entities(
                Payment.id,
```
That's my bad this should all be relative to the payment, the ticket shouldn't factor into this

Reply:
I will update this and push latest changes